src/sidebar/virtual-thread-list.js

```diff
@@ -141,26 +153,39 @@ VirtualThreadList.prototype._updateVisibleThreads = function () {
   for (var i = 0; i < allThreads.length; i++) {
     thread = allThreads[i];
     var threadHeight = this._height(thread.id);
+    var added = false;
 
     if (usedHeight + threadHeight < this.window.pageYOffset - MARGIN_ABOVE) {
       // Thread is above viewport
       offscreenUpperHeight += threadHeight;
     } else if (usedHeight <
       this.window.pageYOffset + visibleHeight + MARGIN_BELOW) {
       // Thread is either in or close to the viewport
-      visibleThreads.push(allThreads[i]);
+      visibleThreads.push(thread);
+      added = true;
     } else {
       // Thread is below viewport
       offscreenLowerHeight += threadHeight;
     }
 
+    // any thread that is not going to go through the render process
+    // because it already fits inside of the viewport should be checked
```

Comment:
> because it already fits inside of the viewport

Presumably you mean "not going to be rendered because it is _outside_ of the viewport"?